src/common/images/thumbnailImage.component.tsx

```diff
@@ -1,34 +1,81 @@
-import { Box } from '@mui/material';
+import { Box, CircularProgress } from '@mui/material';
 import React from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { APIImage } from '../../api/api.types';
 
 export interface ThumbnailImageProps {
   onClick?: () => void;
-  image: APIImage;
+  image?: APIImage;
   dense: boolean;
+  isPrimaryThumbnail?: boolean;
+  imageLoading?: boolean;
 }
 
 const ThumbnailImage = (props: ThumbnailImageProps) => {
-  const { onClick, image, dense } = props;
+  const {
+    onClick,
+    image,
+    dense,
+    isPrimaryThumbnail = false,
+    imageLoading = false,
+  } = props;
   const [hasError, setHasError] = React.useState<string | undefined>(undefined);
+  const imageError = !image || !!hasError;
+
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const handleViewPrimary = React.useCallback(() => {
+    if (isPrimaryThumbnail && image) {
+      const updatedParams = new URLSearchParams(searchParams);
+      updatedParams.set('tab', 'Gallery');
+      updatedParams.set('image', image.id);
+      setSearchParams(updatedParams);
+    }
+  }, [isPrimaryThumbnail, searchParams, setSearchParams, image]);
```

Comment:
This should be moved into primary image component and should be passed in use the onClick prop that already exist in the thumbnail component